test/ScheduledPayment.spec.ts

```diff
@@ -0,0 +1,228 @@
+      const avatar = await avatarFactory.deploy();
+      const ScheduledPayment = await hre.ethers.getContractFactory(
+        "ScheduledPayment"
+      );
+      await expect(
+        ScheduledPayment.deploy(
+          user1.address,
+          AddressZero,
+          avatar.address,
+          AddressZero
+        )
+      ).to.be.revertedWith("Avatar can not be zero address");
+    });
+
+    it("throws if target is zero address", async () => {
+      const avatarFactory = await hre.ethers.getContractFactory("TestAvatar");
+      const avatar = await avatarFactory.deploy();
+      const ScheduledPayment = await hre.ethers.getContractFactory(
+        "ScheduledPayment"
+      );
+      await expect(
+        ScheduledPayment.deploy(
+          user1.address,
+          avatar.address,
+          AddressZero,
+          AddressZero
+        )
+      ).to.be.revertedWith("Target can not be zero address");
+    });
+
+    it("should emit event because of successful set up", async () => {
+      const avatarFactory = await hre.ethers.getContractFactory("TestAvatar");
+      const avatar = await avatarFactory.deploy();
+      const ScheduledPayment = await hre.ethers.getContractFactory(
+        "ScheduledPayment"
+      );
+      const scheduledPayment = await ScheduledPayment.deploy(
+        user1.address,
+        avatar.address,
+        avatar.address,
+        AddressZero
+      );
+      await scheduledPayment.deployed();
+
+      await expect(scheduledPayment.deployTransaction)
+        .to.emit(scheduledPayment, "ScheduledPaymentSetup")
+        .withArgs(
+          user1.address,
+          user1.address,
+          avatar.address,
+          avatar.address,
+          AddressZero
+        );
+    });
+  });
+
+  describe("setConfig", async () => {
+    it("throws if caller not owner", async () => {
+      const { scheduledPayment } = await setupTests();
+      await expect(
+        scheduledPayment.connect(user2).setConfig(AddressOne)
+      ).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("should emit event and update config value", async () => {
+      const { scheduledPayment } = await setupTests();
+      await expect(scheduledPayment.setConfig(AddressOne))
+        .to.emit(scheduledPayment, "ConfigSet")
+        .withArgs(AddressOne);
+      assert.equal(await scheduledPayment.config(), AddressOne);
+    });
+  });
+
+  describe("schedulePayment()", async () => {
+    it("throws if caller not avatar", async () => {
+      const { scheduledPayment } = await setupTests();
+      const spHash = ethers.utils.solidityKeccak256(
+        [
+          "string",
+          "uint256",
+          "string",
+          "uint256",
+          "string",
+          "uint256",
+          "uint256",
+        ],
+        [
+          AddressOne,
+          "1000000000000",
+          AddressOne,
+          "10000000000",
+          AddressOne,
+          "1659072885",
```

Comment:
its unclear what these values represent, could we document that?

Reply:
I will add the comment here

Reply:
added